src/org/opensolaris/opengrok/history/SubversionRepository.java

```diff
@@ -441,23 +451,55 @@ String determineParent() throws IOException {
         try (BufferedReader in = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
             String line;
             while ((line = in.readLine()) != null) {
-                if (line.startsWith("URL:")) {
+                if (line.startsWith(partName)) {
                     String parts[] = line.split("\\s+");
                     if (parts.length != 2) {
                         LOGGER.log(Level.WARNING,
                                 "Failed to get parent for {0}", directoryName);
                     }
-                    parent = parts[1];
+                    part = parts[1];
                     break;
                 }
             }
         }
 
-        return parent;
+        return part;
+    }
+
+    @Override
+    String determineParent() throws IOException {
+        return getInfoPart("URL:");
     }
 
     @Override
-    String determineBranch() {
-        return null;
+    String determineBranch() throws IOException {
+        String branch = null;
+
+        String url = getInfoPart("URL:");
+        int idx;
+        final String branchesStr = "branches/";
+        if ((idx = url.indexOf(branchesStr)) > 0) {
+            branch = url.substring(idx + branchesStr.length());
+        }
+
+        return branch;
+    }
+
+    @Override
+    public String determineCurrentVersion() throws IOException {
+        String curVersion = null;
+
+        try {
+            History hist = getHistory(new File(getDirectoryName()), null, 1);
+            HistoryEntry he = hist.getHistoryEntries().get(0);
+            curVersion = outputDateFormat.format(he.getDate()) + " " +
+                    he.getRevision() + " " + he.getAuthor() + " " +
+                    he.getMessage();
```

Comment:
stringbuilder ?
I am actually not sure if some newer jdks won't already optimize like this so it might be unnecessary ...

Reply:
according to https://stackoverflow.com/questions/1532461/stringbuilder-vs-string-concatenation-in-tostring-in-java it looks like that the compiler will substitute it itself so leaving as is.